api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,44 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  // console.log('bang');
+
+  const token = req.get('x-access-token');
+  if (!token) {
+    throw new ClientError(401, 'authentication required');
+    // console.log('bangbang');
+    // to test, use: http - v get: 3000 / api / flashcards
+    // returns bangbang in the server-side terminal
+    // returns the appropriate response from the server in the client-side httpie terminal
+    // renders in localhost:3000 BROWSER error: authentication required
+    // not exactly sure where/how this is happening
```

Comment:
This `authorizationMiddleware` is being used as a gate keeper in `server/index.js` here

https://github.com/AnthonyVNgo/c0322-code-solutions/pull/129/files#diff-614c0604b7b5624b45327a7250e741ee41f5e63b3699e2619fad19de486fd945R81

Express tries to get routes to handle the request from top to bottom. Since nothing matches just `/` up until this `authorizationMiddleware`, it will intercept the request and send back the `401` response